### ISSUE TITLE
Provide a way to further filter what SuzieQ can match on

### ISSUE THREAD

**Issue by donaldsharp** [netenglabs/suzieq#243]
it would be nice to have suzieq be able to show matched interface pairs across machines

**Comment by jopietsch** [comment]
Can you describe more what you want this to do?

I think you can almost do what you want with topology show:
```
(suzieq-QZqWZG2D-py3.8) jpietsch@jpietsch-server:~/suzieq$ suzieq/cli/suzieq-cli -c ~/configs/multidc.yml topology show --hostname=leaf01 --namespace=ospf-ibgp
    namespace hostname peerHostname         ifname aprnd      vrf   bgp  lldp  ospf  polled_neighbor
0   ospf-ibgp   leaf01       leaf02  peerlink.4094  True      NaN   NaN   NaN   NaN             True
1   ospf-ibgp   leaf01       leaf02         vlan13  True      NaN   NaN   NaN   NaN             True
3   ospf-ibgp   leaf01       leaf02         vlan24  True      NaN   NaN   NaN   NaN             True
5   ospf-ibgp   leaf01       leaf02           swp3   NaN      NaN   NaN  True   NaN             True
6   ospf-ibgp   leaf01       leaf02           swp4   NaN      NaN   NaN  True   NaN             True
7   ospf-ibgp   leaf01      spine01           swp1  True  default  True  True  True             True
28  ospf-ibgp   leaf01      spine02           swp2  True  default  True  True  True             True
49  ospf-ibgp   leaf01    server101      vlan13-v0  True      NaN   NaN   NaN   NaN             True
50  ospf-ibgp   leaf01    server101         vlan13  True      NaN   NaN   NaN   NaN             True
51  ospf-ibgp   leaf01    server102      vlan24-v0  True      NaN   NaN   NaN   NaN             True
52  ospf-ibgp   leaf01    server102         vlan24  True      NaN   NaN   NaN   NaN             True
```

If you do this on your own system, you might get a lot more answers,that are duplicates, I have a one line fix in my local branch that I've not yet moved to master.

**Comment by donaldsharp** [comment]
Talked further w/ Dinesh, SuzieQ needs the ability to add additional filters, something like this:

```topology show namespace=ospf-ibgp hostname=leaf01 ifname=br1```

Conversation from slack:
```
Dinesh G Dutt  8:13 AM
Could you please tell us are you looking for match across physical interface, across logical service (such as routing protocol), something else?

Donald Sharp  8:13 AM
I was really thinking something along the lines of interfaces.

Dinesh G Dutt  8:14 AM
For the classical enterprise EVPN ref arch, here is what the command topology show gives
8:14
ddutt> topology show namespace=ospf-ibgp hostname=leaf01
    namespace hostname peerHostname         ifname aprnd      vrf   bgp  lldp  ospf  polled_neighbor
0   ospf-ibgp   leaf01       leaf02         vlan13  True      NaN   NaN   NaN   NaN             True
2   ospf-ibgp   leaf01       leaf02         vlan24  True      NaN   NaN   NaN   NaN             True
4   ospf-ibgp   leaf01       leaf02  peerlink.4094  True      NaN   NaN   NaN   NaN             True
5   ospf-ibgp   leaf01       leaf02           swp3   NaN      NaN   NaN  True   NaN             True
6   ospf-ibgp   leaf01       leaf02           swp4   NaN      NaN   NaN  True   NaN             True
7   ospf-ibgp   leaf01      spine02           swp2  True  default  True  True  True             True
28  ospf-ibgp   leaf01    server101         vlan13  True      NaN   NaN   NaN   NaN             True
29  ospf-ibgp   leaf01    server101      vlan13-v0  True      NaN   NaN   NaN   NaN             True
30  ospf-ibgp   leaf01    server102      vlan24-v0  True      NaN   NaN   NaN   NaN             True
31  ospf-ibgp   leaf01    server102         vlan24  True      NaN   NaN   NaN   NaN             True
32  ospf-ibgp   leaf01      spine01           swp1  True  default  True  True  True             True
ddutt> 
8:15
There's no LLDP info from server*. Its determined by ARPND. Did you need it broken down further by MAC, instead of ARPND so that you could get the physical interface, if available?

Donald Sharp  8:16 AM
let me take a step back and talk abit about what I was thinking

Dinesh G Dutt  8:16 AM
Thank you

Donald Sharp  8:16 AM
Suppose I have a bridge w/ 3 interfaces in it A, B and C
8:16
Those are connected to machines hosta, hostb and hostc.  Where I am also running suzieQ
8:17
it would be nice to have a command that says Show me the interface names and hosts attached to this bridge
8:17
so not an evpn use case, but probably could be extended
8:19
but explicitly I think it would be nice to grab that data and now that I look at the above command I think it gets it for you, it's just a bunch of data( what happens if we have 40-50 different interfaces )

Dinesh G Dutt  8:19 AM
If you have lotsa data, whether it be interfaces or prefixes, you need to use filters to drill down

Donald Sharp  8:20 AM
yes exactly

Dinesh G Dutt  8:20 AM
The command above gets you the logical connection, not the physical interface

Donald Sharp  8:20 AM
can you do topology show namespace=ospf-ibgp hostname=leaf01 ifname=br1 above?

Dinesh G Dutt  8:21 AM
Nope ```

**Comment by ddutt** [comment]
This specifically applies to the "topology command"

**Comment by ddutt** [comment]
Fixed via commits d645b5552dd8e7c3b6f232bb21d55505785be3bb among others
```
ddutt> topology show namespace=ospf-ibgp hostname=leaf01 ifname=swp1
   namespace hostname ifname peerHostname      vrf  arpnd  arpndBidir   bgp  lldp  ospf  polled
0  ospf-ibgp   leaf01   swp1      spine01  default   True        True  True  True  True    True
ddutt> 
```
